src/js/components/Header/Header.js

```diff
@@ -1,19 +1,141 @@
-import React from 'react';
-
+import React, { useContext, useEffect, useState } from 'react';
+import { ThemeContext } from 'styled-components';
+import { PageContext } from '../Page';
+import { useForwardedRef } from '../../utils';
 import { Box } from '../Box';
 
-const Header = React.forwardRef(({ ...rest }, ref) => (
-  <Box
-    align="center"
-    as="header"
-    direction="row"
-    flex={false}
-    justify="between"
-    gap="medium"
-    {...rest}
-    ref={ref}
-  />
-));
+const Header = React.forwardRef(({ sticky, background, ...rest }, ref) => {
+  const theme = useContext(ThemeContext);
+  const containerRef = useForwardedRef(ref);
+  const [stickyStyles, setStickyStyles] = useState();
+  const { ...pageContext } = useContext(PageContext);
+
+  useEffect(() => {
+    // threshold inital value set to 0
+    const threshold = 0;
+    // number of pixels the document is currently scrolled
+    let lastScrollY = window.pageYOffset;
+    // run our event listener callback once in each requestAnimationFrame
+    let runEventListener = false;
+
+    const updateScrollDir = () => {
+      const scrollY = window.pageYOffset;
+      if (Math.abs(scrollY - lastScrollY) < threshold) {
+        runEventListener = false;
+        return;
+      }
+      if (scrollY === 0) {
+        runEventListener = false;
+        setStickyStyles(undefined);
+      }
+      if (scrollY < lastScrollY) {
+        setStickyStyles({
+          width: `${containerRef.current.getBoundingClientRect().width}px`,
+          height: `${containerRef.current.getBoundingClientRect().height}px`,
+          left: `${containerRef.current.getBoundingClientRect().left}px`,
+          right: `${containerRef.current.getBoundingClientRect().right}px`,
+          zIndex: `${theme.header?.sticky?.zIndex}`,
+          position: 'fixed',
+          transition: 'top 0.6s',
+          top: '0',
+        });
+      } else
+        setStickyStyles({
+          top: `-${containerRef.current.getBoundingClientRect().height}px`,
```

Comment:
I feel like sometimes when I'm scrolling down, I don't see the animation. Do we need to add the transition back here?

Reply:
oh really Ill take a look

Reply:
i've attached a video, notice when I scroll down how the header abruptly disappears. In order to see this, you have to be scrolled down the page a bit (not right at the top).

https://user-images.githubusercontent.com/12522275/161622911-fc6d478c-cff3-4d3d-b025-8a517d3196ae.mov

Reply:
And then just a general question, is the position still "fixed" when we scroll down or what does `top` do we we don't have position fixed?

Reply:
I updated this let me know what you think